### PR TITLE
CPDT-213 B-Button significant updates

### PR DESCRIPTION
**Pull request by eleanormal**
## Type of change

- [ ] Bug Fix
- [x] New Feature
- [ ] Refactor
- [ ] Project Build
- [ ] Project Configuration

## Submission Checklist

- [x] Have you followed the guidelines in our [Contributing document](https://github.com/fanappics/buoy/blob/master/CONTRIBUTING.md)?
- [x] Did you create appropriate tests?

## Description

Updated `b-button` style and patterns including:
Add `b-button` class and prefix everything with `b-`
Use `label` property to match what we do in other places (update related examples)
Slot is now only for passing in an icon
Move all the `b-button` styles into that vue file (leftover button style is for `b-select` which is next on my list)
Create primary, secondary, tertiary styles to match Christian's designs
Removed the computed style and class functions that were future-optimizations we don't know we'll need
Added lots of examples and documentation

## Related Issues / Bugs / Feature Requests

https://jira.fanatics.com/browse/CPDT-213

## Test Deployment URL

Replace branch-name with the name of your branch.

https://rawgit.com/fanappics/buoy/ft-CPDT-213/docs/index.html#bbutton

## Deployment Notes

N/A

## Steps to Test or Reproduce

Compare with https://app.zeplin.io/project/5a7a2d4c69fae4776dce32dd/screen/5a97250b08be6be74579242b

### CODE REVIEW

File: src/styles/colors.styl [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
+$ironside-gray = #646261
 $fedora = #6B686B
 $mountain-mist = #A1A0A2
 $silver = #C1C1C1
 $pale-slate = #c4c3c4
 $venetian-red = #D0021B
 $iron = #dde0e0
 $quartz = #DEDEDF
-$athens-grey = #f5f6f8
+$athens-gray = #f5f6f8
```
Comment (mlear) [inline src/styles/colors.styl]: -1 gray.  grey for life!

**Comment by kierachell** [comment]
- Seems like the button's size changes on click.
- Disabled state still changes color on click: intended?
dat_page_shift

File: src/styles/colors.styl (hunk 1 above)
Comment (eleanormal) [reply src/styles/colors.styl]: I think the rule is "gray" in **A**merica and "grey" in **E**ngland. And "grehy" in Canad**eh**

**Comment by eleanormal** [comment]
The disabled one seems like a FF bug because you shouldn't be able to set focus on a disabled button but I'll set the disabled style more thoroughly.

Also - stupid inset border is what's causing the weirdness. I shall fix.

**Comment by eleanormal** [comment]
Also, there is a style change on focus (inset shadow), but no outline because it blends in with the color of the primary button.

**Pull request merged**